apps/prairielearn/src/pages/instructorAssessmentAccessOverrides/instructorAssessmentAccessOverrides.js

```diff
@@ -0,0 +1,165 @@
+// @ts-check
+import * as express from 'express';
+const asyncHandler = require('express-async-handler');
+import * as sqldb from '@prairielearn/postgres';
+import * as error from '@prairielearn/error';
+import { getEnrollmentForUserInCourseInstance } from '../../models/enrollment';
+import { selectUserByUid } from '../../models/user';
+
+const router = express.Router();
+const sql = sqldb.loadSqlEquiv(__filename);
+
+async function userIsEnrolledInCourseInstance({ uid, course_instance_id }) {
+  const user = await selectUserByUid(uid);
+  if (!user) return false;
+
+  const enrollment = await getEnrollmentForUserInCourseInstance({
+    user_id: user.user_id,
+    course_instance_id: course_instance_id,
+  });
+  return !!enrollment;
+}
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    const result = await sqldb.queryAsync(sql.select_assessment_access_policies, {
+      assessment_id: res.locals.assessment.id,
+      student_uid: req.body.student_uid,
+      course_instance_id: res.locals.course ? res.locals.course_instance.id : null,
+      course_id: res.locals.course ? res.locals.course.id : null
+    });
+    res.locals.policies = result.rows;
+    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    
+    if (req.body.__action === 'add_new_override') {
+      const isEnrolled = await userIsEnrolledInCourseInstance({
+        uid: req.body.student_uid,
+        course_instance_id: res.locals.course_instance.id,
+      });
+      if (!isEnrolled) {
+        throw error.make(400, `User ${req.body.student_uid} is not enrolled in this course instance.`);
+      }
+      const user = await selectUserByUid(req.body.student_uid);
```

Comment:
Think through what happens here for a group assessment when `req.body.student_uid` won't be present. Did you test this code? Ditto for `edit_override` below.

To start, ask yourself why `userIsEnrolledInCourseInstance` is called twice here.

Reply:
705f836e4 abstracts further, but do you mean I should call getUserIdAndCheckEnrollment before the conditionals checking what the action for the override is

Reply:
I think the changes in 705f836 are a regression; recall our previous discussion here: https://github.com/PrairieLearn/PrairieLearn/pull/9103#discussion_r1496735902

I'm not suggesting that you need to do these checks before you check `req.body.__action`. You've resolved my above comments with your recent changes, just make sure to consider the regression I mentioned.